server/routes/induction/update/affectAbilityToWorkFormValidator.test.ts

```diff
@@ -0,0 +1,84 @@
+import type { AffectAbilityToWorkForm } from 'inductionForms'
+import validateAffectAbilityToWorkForm from './affectAbilityToWorkFormValidator'
+import aValidPrisonerSummary from '../../../testsupport/prisonerSummaryTestDataBuilder'
+
+describe('affectAbilityToWorkFormValidator', () => {
+  const prisonerSummary = aValidPrisonerSummary()
+
+  describe('happy path - validation passes', () => {
+    Array.of<AffectAbilityToWorkForm>(
+      { affectAbilityToWork: ['OTHER'], affectAbilityToWorkOther: 'Mental health' },
+      { affectAbilityToWork: ['NO_RIGHT_TO_WORK'], affectAbilityToWorkOther: '' },
+      { affectAbilityToWork: ['NO_RIGHT_TO_WORK'], affectAbilityToWorkOther: undefined },
+      { affectAbilityToWork: ['NO_RIGHT_TO_WORK', 'OTHER'], affectAbilityToWorkOther: 'Mental health' },
+      { affectAbilityToWork: ['NONE'], affectAbilityToWorkOther: '' },
+    ).forEach(spec => {
+      it(`form data: ${JSON.stringify(spec)}`, () => {
+        // Given
+        const expected: Array<Record<string, string>> = []
+
+        // When
+        const actual = validateAffectAbilityToWorkForm(spec, prisonerSummary)
+
+        // Then
+        expect(actual).toEqual(expected)
+      })
+    })
+  })
+
+  //  //  LIMITED_BY_OFFENSE = 'LIMITED_BY_OFFENSE',
+  //   //   CARING_RESPONSIBILITIES = 'CARING_RESPONSIBILITIES',
+  //   //   HEALTH_ISSUES = 'HEALTH_ISSUES',
+  //   //   NO_RIGHT_TO_WORK = 'NO_RIGHT_TO_WORK',
+  //   //   OTHER = 'OTHER',
+  //   //   NONE = 'NONE',
+  describe('sad path - validation of affectAbilityToWork field does not pass', () => {
+    Array.of<AffectAbilityToWorkForm>(
+      { affectAbilityToWork: [], affectAbilityToWorkOther: 'Mental health' },
+      { affectAbilityToWork: [], affectAbilityToWorkOther: '' },
+      { affectAbilityToWork: [], affectAbilityToWorkOther: undefined },
+      { affectAbilityToWork: ['a-non-supported-value'], affectAbilityToWorkOther: undefined },
+      { affectAbilityToWork: ['NO_RIGHT'], affectAbilityToWorkOther: undefined },
+      { affectAbilityToWork: ['NO_RIGHT_TO_WORK', 'a-non-supported-value'], affectAbilityToWorkOther: undefined },
+      { affectAbilityToWork: ['NO_RIGHT_TO_WORK', 'NONE'], affectAbilityToWorkOther: undefined },
+    ).forEach(spec => {
+      it(`form data: ${JSON.stringify(spec)}`, () => {
+        // Given
+        const expected: Array<Record<string, string>> = [
+          {
+            href: '#affectAbilityToWork',
+            text: `Select factors affecting Jimmy Lightfingers's ability to work or select 'None of these'`,
```

Comment:
One day we'll fix this (but not in these PRs)

ie. the incorrect placement of the apostrophe when surnames end in an `s` -  in this case it shouldn't it be `Jimmy Lightfingers' ability to work` ? 😁 :trollface:

Reply:
Controversial! I think both are correct actually: https://www.wikihow.com/James-or-James%27s